tests/sp2n/Test_project_on_Sp.cc

```diff
@@ -17,27 +17,32 @@ int main (int argc, char **argv)
     
     LatticeGaugeField Umu(&Grid);
     LatticeColourMatrixD U(&Grid);
+    LatticeColourMatrixD Up(&Grid);
     LatticeColourMatrixD aux(&Grid);
     LatticeColourMatrixD identity(&Grid);
     
+    // Will test resimplectification-related functionalities (from ProjectOnGaugeGroup, ProjectOnSpGroup, ProjectGn) and projection on the algebra (from ProjectSp2nAlgebra)
+    // we work with matrices with positive determinant so detU = 1 even if in principle ProjectOnGaugeGroup and ProjectOnSpGroup allow for detU=-1
+    // so the checks will be the same for the three functions
+    // NB only ProjectGn is the proper simplectification function
```

Comment:
What for do the other two exists then? If only for internal purposes, should they be private?